src/core/dbus-manager.c

```diff
@@ -1471,6 +1520,10 @@ static int method_switch_root(sd_bus_message *message, void *userdata, sd_bus_er
         Manager *m = userdata;
         int r;
 
+        r = verify_run_space("Refusing to switch root", error);
+        if (r < 0)
+                return r;
```

Comment:
I don't think that's good. For all intents and purposes, failing to switch root is as good as crashing. And since we don't really know how much space we're going to need, I think here this should only be a warning ("Dangerously low amount of free space on /var, we might crash" or so). In particular, for embedded systems with really low amounts of memory this check might be unnecessarily punitive.